fedimint-server/src/config.rs

```diff
@@ -64,52 +77,152 @@ pub struct Peer {
     pub name: String,
 }
 
+pub struct FederationeConfigGenParams {
+    pub mint_amounts: Vec<Amount>,
+    pub bitcoin_rpc: BitcoindRpcCfg,
+
+    /// extra options for extra settings and modules
+    pub other: BTreeMap<String, serde_json::Value>,
+}
```

Comment:
This will be directly in conflict with modularization. Why not take a similar approach to the server config and have a subsection per module?

Reply:
RN `mint_amounts` and `bitcoin_rpc` seem to be all that is necessary for configgen. This `other` is currently not used, and I'll probably delete it, but when I was thinking about it it seemed to me that for configgen we are not really going to need a config-per-module, but a one global config from which modules can pick and use what they need. Since we can't predict every aspect of a potential module, we need an escape hatch, but I don't think it necessarily has to be `module -> config` mapping.

Reply:
Ok, ignore my other comment regarding this then. What's the reason for the code duplication with https://github.com/fedimint/fedimint/pull/902/files#diff-088558228243c1935ac45a97aece30fa637c0f82d842732a9039fc1040c5b9beR65-R71